File: JavaScript/프로그래머스/2/42584. 주식가격/주식가격.js
function solution(prices) {
    
    let stack = [];
    const len = prices.length;
    const answer = new Array(len).fill(0);
    for (let i = 0; i < len; i++){
        if (i === len - 1){
            break;
        }
        if (stack.length === 0)
            stack.push([i, prices[i]]); // 위치와 값
        else {
            while (stack.length !== 0 && stack[stack.length - 1][1] > prices[i]){
                const [a, b]  = stack.pop();
                answer[a] = i - a;
            }   
            stack.push([i, prices[i]]);
        }
    }
    while (stack.length !== 0){
        const [a, b] = stack.pop();
        answer[a] = len - 1 - a;
    }
    
    return answer;
}

File: C++/프로그래머스/2/42584. 주식가격/주식가격.cpp
#include <string>
#include <vector>
#include <utility>

using namespace std;

vector<int> solution(vector<int> prices) {
    vector<int> answer(prices.size(), 10);
    vector<pair<int, int>> stk;
    for (int i = 0; i < prices.size(); i++){
        if (stk.empty()) stk.push_back({prices[i], i});
        else {
            int temp = prices[i];
            while (!stk.empty() && stk.back().first > temp){ // 스택이 차있고
                answer[stk.back().second] = i - stk.back().second; 
                stk.pop_back();
            }
            stk.push_back({prices[i], i});
        }
    }
    while (!stk.empty()){
        answer[stk.back().second] = prices.size() - 1 - stk.back().second;
        stk.pop_back();
    }
    return answer;
}